test/SwapVar.lean

```diff
@@ -0,0 +1,22 @@
+/-
+Copyright (c) 2022 Arthur Paulino. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino
+-/
+
+import Mathlib.Tactic.SwapVar
+
+example {P Q : Prop} (q : P) (p : Q) : P ∧ Q := by
+  swap_var p ↔ q
+  exact .intro p q
+
+example {a b : Nat} (h : a = b) : a = b ∧ a = a := by
+  swap_var a ↔ b
+-- b a: Nat
+-- h: b = a
+-- ⊢ b = a ∧ b = b
+  swap_var b y, y ↔ x
+-- x a: Nat
+-- h: x = a
+-- ⊢ x = a ∧ x = x
+  exact .intro h $ .refl x
```

Comment:
```suggestion
example {P Q : Prop} (q : P) (p : Q) : P ∧ Q := by
  swap_var p ↔ q
  exact ⟨p, q⟩

example {a b : Nat} (h : a = b) : a = b ∧ a = a := by
  swap_var a ↔ b
  guard_hyp h : b = a
  guard_target == b = a ∧ b = b
  exact ⟨h, Eq.refl b⟩

example {a b c d : Nat} (h : a = b ∧ c = d) : a = b ∧ c = d := by
  swap_var a ↔ b, b c
  guard_target == c = a ∧ b = d
  exact h
```
Removed the test with `y` because the lean 3 semantics throws an error when a variable in the cycle is not present in the context.